openshift-kube-apiserver/admission/autoscaling/managementcpusoverride/admission.go

```diff
@@ -180,17 +181,23 @@ func (a *managementCPUsOverride) Admit(ctx context.Context, attr admission.Attri
 		return admission.NewForbidden(attr, err) // can happen due to informer latency
 	}
 
-	// we can not decide the cluster type without status.controlPlaneTopology and status.infrastructureTopology
-	if clusterInfra.Status.ControlPlaneTopology == "" || clusterInfra.Status.InfrastructureTopology == "" {
-		return admission.NewForbidden(attr, fmt.Errorf("%s infrastructure resource has empty status.controlPlaneTopology or status.infrastructureTopology", PluginName))
+	// the infrastructure status is empty, so we can not decide the cluster type
+	if reflect.DeepEqual(clusterInfra.Status, configv1.InfrastructureStatus{}) {
+		return admission.NewForbidden(attr, fmt.Errorf("%s infrastructure resource has empty status", PluginName))
+	}
+
+	// the infrastructure status is not empty, but topology related fields do not have any values indicates that
+	// the cluster is during the roll-back process to the version that does not support the topology fields
+	// the upgrade to 4.8 handled by the CR defaulting
+	if clusterInfra.Status.ControlPlaneTopology == "" && clusterInfra.Status.InfrastructureTopology == "" {
```

Comment:
you don't actually need this block, right?  Because if they are empty strings for whatever reason, the `!= configv1.SingleReplicaTopologyMode` escape immediately below will already `return nil`.

Reply:
I prefer an explicit statement rather than a coincidence in code that does not intentionally handles this case.